updatem/orchestration/update_orchestrator_apply_test.go

```diff
@@ -0,0 +1,422 @@
+			expectedStatus: types.StatusIdentifying,
+		},
+		"test_case_done": {
+			ctx:            context.Background(),
+			testChan:       done,
+			expectedErr:    nil,
+			expectedStatus: types.StatusIdentifying,
+		},
+		"test_case_terminateContext": {
+			ctx:              context.Background(),
+			testChan:         none,
+			expectedErr:      fmt.Errorf("the update manager instance is terminated"),
+			expectedStatus:   types.StatusIncomplete,
+			terminateContext: true,
+		},
+	}
+
+	for testName, testCase := range testCases {
+		t.Run(testName, func(t *testing.T) {
+			orchestrator := &updateOrchestrator{
+				operation: &updateOperation{
+					errChan: make(chan bool, 1),
+					done:    make(chan bool, 1),
+					errMsg:  "testErrMsg",
+					status:  types.StatusIdentifying,
+				},
+			}
+
+			if testCase.testChan == done {
+				orchestrator.operation.done <- true
+			} else if testCase.testChan == errChan {
+				orchestrator.operation.errChan <- true
+			}
+
+			var actualErr error
+			if testCase.terminateContext {
+				newContext, cancel := context.WithTimeout(testCase.ctx, time.Second)
+				cancel()
+				actualErr = orchestrator.waitCompletion(newContext)
+			} else {
+				actualErr = orchestrator.waitCompletion(testCase.ctx)
+			}
+
+			assert.Equal(t, testCase.expectedErr, actualErr)
+			assert.Equal(t, testCase.expectedStatus, orchestrator.operation.status)
+		})
+	}
+}
+
+func TestCommand(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockUpdateManager := mocks.NewMockUpdateManager(mockCtrl)
+
+	t.Run("test_command_existing_domain", func(t *testing.T) {
+		orchestrator := &updateOrchestrator{
+			operation: &updateOperation{},
+		}
+		orchestrator.operation.statesPerDomain = map[api.UpdateManager]*types.DesiredState{
+			mockUpdateManager: {},
+		}
+		command := &types.DesiredStateCommand{
+			Command: types.CommandActivate,
+		}
+
+		mockUpdateManager.EXPECT().Name().Return("testName").Times(1)
+		mockUpdateManager.EXPECT().Command(context.Background(), test.ActivityID, command)
+
+		orchestrator.command(context.Background(), test.ActivityID, "testName", types.CommandActivate)
+	})
+	t.Run("test_command_domain_not_exists", func(t *testing.T) {
+		orchestrator := &updateOrchestrator{
+			operation: &updateOperation{},
+		}
+		orchestrator.operation.statesPerDomain = map[api.UpdateManager]*types.DesiredState{
+			mockUpdateManager: {},
+		}
+
+		mockUpdateManager.EXPECT().Name().Return("testName").Times(1)
+
+		orchestrator.command(context.Background(), test.ActivityID, "difTestName", types.CommandActivate)
+	})
+}
+
+func TestSetupUpdateOperation(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	t.Run("test_setupupdateoperation_newUpdateOperation_err_nil", func(t *testing.T) {
+
+		domainAgents := map[string]api.UpdateManager{
+			"domain1": mocks.NewMockUpdateManager(mockCtrl),
+		}
+		activityID := test.ActivityID
+		desiredState := &types.DesiredState{
```

Comment:
The same desired state is repeated three times in the project. Could you move it to the constants, please?